src/components/DisplayObject.js

```diff
@@ -0,0 +1,133 @@
+import invariant from 'fbjs/lib/invariant'
+import { DisplayObject } from 'pixi.js'
+import { PROPS_DISPLAY_OBJECT, PROPS_RESERVED } from '../utils/props'
+import { isPointType, parsePoint } from '../utils/pixi'
+import { not, hasKey } from '../helpers'
+
+DisplayObject.getReactEventHandlers = function() {
+  return DisplayObject._reactEventHandlers
+}
+
+DisplayObject.setReactEventHandlers = function(eventHandlers) {
+  Object.assign(DisplayObject, {
+    _reactEventHandlers: eventHandlers,
+    _reactFilterProps: not(hasKey([...Object.keys(PROPS_RESERVED), ...eventHandlers])),
+  })
+}
+
+DisplayObject.setReactEventHandlers([
+  'click',
+  'mousedown',
+  'mousemove',
+  'mouseout',
+  'mouseover',
+  'mouseup',
+  'mouseupoutside',
+  'tap',
+  'touchstart',
+  'touchmove',
+  'touchend',
+  'touchendoutside',
+  'pointercancel',
+  'pointerout',
+  'pointerover',
+  'pointertap',
+  'pointerdown',
+  'pointerup',
+  'pointerupoutside',
+  'pointermove',
+  'rightclick',
+  'rightdown',
+  'rightup',
+  'rightupoutside',
+  'touchcancel',
+  'touchendoutside',
+  'touchmove',
+  'touchstart',
+])
+
+Object.assign(DisplayObject.prototype, {
+  _reactSetValue: function(prop, value) {
+    const instance = this
+    if (isPointType(instance[prop]) && isPointType(value)) {
+      // copy value
+      instance[prop].copyFrom(value)
+    } else if (isPointType(instance[prop])) {
+      // parse value if a non-Point type is being assigned to a Point type
+      const coordinates = parsePoint(value)
+
+      invariant(
+        typeof coordinates !== 'undefined' && coordinates.length > 0 && coordinates.length < 3,
+        'The property `%s` is a `PIXI.Point` or `PIXI.ObservablePoint` and must be set to a comma-separated string of ' +
+          'either 1 or 2 coordinates, a 1 or 2 element array containing coordinates, or a PIXI Point/ObservablePoint. ' +
+          'If only one coordinate is given then X and Y will be set to the provided value. Received: `%s` of type `%s`.',
+        prop,
+        JSON.stringify(value),
+        typeof value
+      )
+
+      instance[prop].set(coordinates.shift(), coordinates.shift())
+    } else {
+      // just hard assign value
+      instance[prop] = value
+    }
+  },
+  _reactApplyProps: function(oldProps, newProps) {
```

Comment:
why the underscore? The applyProps is something used quite often. I suggest removing the underscore, so you can use it like: `DisplayObject.prototype.reactApplyProps`

Reply:
I think the underscore is used for private methods on Pixi.js library.

Reply:
we're using `reactApplyProps` also publicly, let's get rid of the underscore. Are you ok with that?

Reply:
Sure! :)

Reply:
Can u fix conflicts?

Reply:
@eXponenta I will try to fix this when I have time. My client is hammering me right now...